cpp/src/arrow/engine/substrait/options.cc

```diff
@@ -166,6 +171,57 @@ class DefaultExtensionProvider : public BaseExtensionProvider {
                                      named_tap_rel.name(), std::move(renamed_schema)));
     return RelationInfo{{std::move(decl), std::move(renamed_schema)}, std::nullopt};
   }
+
+  Result<RelationInfo> MakeSegmentedAggregateRel(
+      const ConversionOptions& conv_opts, const std::vector<DeclarationInfo>& inputs,
+      const substrait_ext::SegmentedAggregateRel& seg_agg_rel,
+      const ExtensionSet& ext_set) {
+    if (inputs.size() != 1) {
```

Comment:
I think this is no longer a concern right?  Since we are only using the `Measure` part of the original message (and not the part that contains its own inputs).  So we should be able to avoid this `inputs` check.

Reply:
I think we still need this check, as the inputs are coming from the caller, not from the `SegmentedAggregateRel` message. We have similar code in the existing `MakeAsOfJoinNodeRel`.

Reply:
This is probably OK to keep (minor point)

Reply:
I get it now.  I think, in the past, we had different branches for `kExtensionLeaf`, `kExtensionSingle`, and `kExtensionMulti`.  In my brain this was a `kExtensionSingle` only and so we would verify the number of inputs there.

However, we've merged those all into one path in the extension provider, so we still get `inputs` and need to verify the length.

+1 for leaving this in, sorry for the confusion.